### DISCUSSION TITLE
Library for package managers independent command execution

### DISCUSSION THREAD

**Discussion by flying-sheep** [yarnpkg/berry#4051] [Q&A]
So in order to run a command from a package installed into the workspace, there’s `yarn run <cmd>`, `npx --no <cmd>`, and probably others.

Since yarn 2 doesn’t (by default) install things into `node_modules`, we now lack a cross-package-manager way to run a locally installed command (before yarn 2, that was `<workspace-path>/node_modules/.bin/<cmd>`)

Is there a library that handles this? I.e. figures out if the current workspace is managed by yarn 2 and uses `yarn run` then, and the equivalent other commands for other package managers?

**Comment by arcanis** [comment]
You can just call `<bin>`, since all package managers add it into the PATH.

**Reply by flying-sheep** [reply]
Nonono, what makes you think that?

They do that precisely when you use their package manager specific run mechanism, otherwise not. Two example use cases where that’s not the case:

- A tool could have been launched simply via `absolute/path/to/tool`
- When a tool wants control over which workspace to execute something in, it can’t use `$PATH`
